contrib/python/pypodman/pypodman/lib/parser_actions.py

```diff
@@ -94,21 +91,40 @@ def __init__(self,
             help=help,
             metavar=metavar)
 
-    def __call__(self, parser, namespace, values, option_string=None):
-        """Convert and Validate input."""
-        items = getattr(namespace, self.dest, None) or []
-        items = copy.copy(items)
+        if hasattr(signal, "Signals"):
 
-        choices = ('CMD', 'ENTRYPOINT', 'ENV', 'EXPOSE', 'LABEL', 'ONBUILD',
-                   'STOPSIGNAL', 'USER', 'VOLUME', 'WORKDIR')
+            def _signal_number(signame):
+                cooked = 'SIG{}'.format(signame)
+                try:
+                    return signal.Signals[cooked].value
+                except ValueError:
+                    pass
+        else:
 
-        opt, val = values.split('=', 1)
-        if opt not in choices:
-            parser.error('Option "{}" is not supported by argument "{}",'
-                         ' valid options are: {}'.format(
-                             opt, option_string, ', '.join(choices)))
-        items.append(values)
-        setattr(namespace, self.dest, items)
+            def _signal_number(signame):
+                cooked = 'SIG{}'.format(signame)
+                for n, v in sorted(signal.__dict__.items()):
+                    if n != cooked:
+                        continue
+                    if n.startswith("SIG") and not n.startswith("SIG_"):
+                        return v
+
+        self._signal_number = _signal_number
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        """Validate input is a signal for platform."""
+        if values.isdigit():
+            signum = int(values)
+            if signal.SIGRTMIN <= signum >= signal.SIGRTMAX:
```

Comment:
That makes my head hurt, and I don't actually think it works as intended. What would you think of this instead?
```python
    if not(1 <= signum <= signal.SIGRTMAX):
```
(replacing `SIGRTMIN` in the diagnostic as well)?

Reply:
You are correct SGRT* were not the values I assumed.  Updating to
```if not(1 <= signum < signal.NSIG):```